### DISCUSSION TITLE
How should I implement a multi-select dropdown box? It seems that the only one available now is the [Select] single choice one.

### DISCUSSION THREAD

**Discussion by gkj17** [material-components/material-web#5578] [Q&A]
How should I implement a multi-select dropdown box? It seems that the only one available now is the [[Select] ](https://github.com/material-components/material-web/blob/main/docs/components/select.md)single choice one.

**Comment by asyncliz** [comment]
Our select component implements the [aria combobox select-only pattern](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/), which is single-select.

A multi-select component is much more complicated, but the tl;dr would be to use an [aria menu button pattern](https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/) and an [aria multi-select listbox](https://www.w3.org/WAI/ARIA/apg/patterns/listbox/).

With our components, you can build this using [field](https://github.com/material-components/material-web/tree/main/field) (so it looks like a select/text field), [menu's surface position controller](https://github.com/material-components/material-web/blob/main/menu/internal/controllers/surfacePositionController.ts) (for the pop-up), and a customized [list with list items](https://github.com/material-components/material-web/tree/main/list) (and aria attributes on them to indicate multi-selection).

It's not an easy component to build, which is why we don't have it yet 😔
